modules/python/test/test_misc.py

```diff
@@ -50,39 +68,538 @@ class Arguments(NewOpenCVTests):
 
+    def test_parse_to_bool(self):
+        def try_to_convert(value):
+            try:
+                result = cv.utils.dumpBool(value).lower()
+            except Exception as e:
+                self.fail(
+                    '{} "{}" is risen for conversion {} of type {}'.format(
+                        type(e).__name__, e, value, type(value).__name__
+                    )
+                )
+            else:
+                return result
+
+        for convertible_true in (True, 1, 64, np.bool(1), np.int8(123),
+                                 np.int16(11), np.int32(2), np.int64(1)):
+            actual = try_to_convert(convertible_true)
+            self.assertEqual('bool: true', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: true' vs Actual: {2}".format(
+                                 convertible_true,
+                                 type(convertible_true).__name__,
+                                 actual
+                             ))
+
+        for convertible_false in (False, 0, np.uint8(0)):
+            actual = try_to_convert(convertible_false)
+            self.assertEqual('bool: false', actual,
+                             msg="Can't convert {0} of type {1}\n"
+                             "Expected: 'bool: false' vs Actual: {2}".format(
+                                 convertible_false,
+                                 type(convertible_false).__name__,
+                                 actual
+                             ))
+
+        a = np.array([1, 2, 0], dtype=np.bool)
+        for i in range(len(a)):
+            expected = 'Bool: {}'.format(bool(a[i])).lower()
+            actual = try_to_convert(a[i])
+            self.assertEqual(expected, actual,
+                             msg="Can't convert array element {0} of type {1}\n"
+                             "Expected: {2} vs Actual: {3}".format(
+                                 a[i],
+                                 type(a[i]).__name__,
+                                 expected, actual
+                             ))
+
+        with self.assertRaises(
+            TypeError,
+            msg='Exception is not risen for tuple slice with 1 element'
+        ):
+            _ = cv.utils.dumpBool((True, False, False)[0:1])
+
+        for not_convertible in (1.2, np.float(2.3), 's', 'str', (1, 2),
+                                [1, 2], a, complex(1, 1), complex(imag=2),
+                                complex(1.1), None):
+            with self.assertRaises(
+                (TypeError, OverflowError),
+                msg='Exception is not risen for {} of type {}'.format(
+                    not_convertible, type(not_convertible).__name__
+                )
+            ):
+                _ = cv.utils.dumpBool(not_convertible)
+
+    @unittest.skip('Wrong conversion behavior')
```

Comment:
What is the plan for skipped tests: are we going to fix them? (is it possible?) Or maybe we should declare these cases unsupported and move them to negative tests?

Reply:
Plan is to fix them with patch from here: #15915 (should be enabled there)

These tests purpose is ensuring that we don't break something else.